Angular view subgenerator and test added

UndeadBaneGitHub opened the pull request.

codydaig commented on angular-view/templates/_.client.route.js:
```diff
@@ -0,0 +1,5 @@
+	state('<%= slugifiedName %>', {
+		url: '/<%= slugifiedRoutePath %>',
+		templateUrl: 'modules/<%= slugifiedModuleName %>/views/<%= slugifiedName %>.client.view.html'
+	}).
```
New Line

codydaig commented on angular-view/templates/_.client.route.js:
```diff
@@ -0,0 +1,5 @@
+$stateProvider.
+	state('<%= slugifiedName %>', {
```
Spacing to 2 spaces.

codydaig commented on angular-view/templates/_.client.routes.js:
```diff
@@ -0,0 +1,13 @@
+
+//Setting up route
+angular.module('<%= slugifiedModuleName %>').config(['$stateProvider',
+	function($stateProvider) {
```
Spacing to 2 spaces.

codydaig commented on angular-view/templates/_.client.view.html:
```diff
@@ -0,0 +1,3 @@
+<% if(controllerName) { %><section data-ng-controller="<%= classifiedControllerName %>Controller"><% } else { %><section><% } %>
+	This is the <%= humanizedName %>
+</section>
```
New Line.

codydaig commented:
Left some line comments.
Don't forget to squash.

Otherwise, LGTM.

codydaig commented:
@ilanbiala LGTY?

ilanbiala commented on angular-view/templates/_.client.route.js:
```diff
@@ -0,0 +1,5 @@
+  state('<%= slugifiedName %>', {
+    url: '/<%= slugifiedRoutePath %>',
+    templateUrl: 'modules/<%= slugifiedModuleName %>/client/views/<%= slugifiedName %>.client.view.html'
+  }).
```
newline?

ilanbiala commented:
@UndeadBaneGitHub Same thing here with working with existing files, can you add a test that actually mimics the behaviors we can expect?

UndeadBaneGitHub commented:
@ilanbiala I think, I will add a test today, that first creates a routes file, and then adds additional route to it. 
The only problem is, that to test actual code insertion for the route I would have to include a code snippet to test against into the body of the test. This code snippet would have to be updated later on if the template changes, which is sorta drag.

ilanbiala commented:
Yeah, I mean that's a little annoying, but we can copy most of it straight from a MEAN file, no need to write one from scratch.

UndeadBaneGitHub commented:
@ilanbiala @codydaig Merged the latest master in and updated the tests to test actual route generation, when a file already exists.
Turned out to be a really good thing, as I caught a yo deprecation-caused bug in this routine and fixed it.
Please, check it out, and if it LGTY, then I will add likewise tests to the two remaining PRs and squash this commit. Don't want to add them beforehand and make a mess of code.

Also, to resolve the conflicts, I found nothing better, than merging the actual master with its `gulp test` into the branch. So, there are quite a few files added, but the relevant ones are prefixed as `angular-view` or sitting in a folder with this name.

And I fixed gulp file definition to exclude `./test/temp` folder and to have explicitly declared arrays as `.src` parameters - otherwise, exclusion did not seem to work.
I will fix this temp folder together with `timeout(0);` everywhere, but for now, I just excluded it from gulp loader.

ilanbiala commented on angular-view/index.js:
```diff
@@ -0,0 +1,138 @@
+        var routesFileContent = this.readFileAsString(routesFilePath);
+
+        // Append the new state
+        routesFileContent = routesFileContent.replace('$stateProvider.', engine(this.read('_.client.route.js'), this));
```
Why do you need ejs?

UndeadBaneGitHub replied:
Before, 'this.engine' was used (that very same 'ejs') for rendering
template into a string. Now, in later version of 'yo', 'this.engine' was
removed, so I added it as dependency.

On Friday, 8 January 2016, Ilan Biala <notifications@github.com> wrote:

> In angular-view/index.js
> <https://github.com/meanjs/generator-meanjs/pull/165#discussion_r49158473>
> :
>
> > +        done();
> > +      }.bind(this));
> > +    }
> > +  },
> > +
> > +  renderRoute: function () {
> > +    if (this.addRoute) {
> > +      var routesFilePath = process.cwd() + '/modules/' + this.slugifiedModuleName + '/client/config/' + this.slugifiedModuleName + '.client.routes.js';
> > +
> > +      // If routes file exists we add a new state otherwise we render a new one
> > +      if (fs.existsSync(routesFilePath)) {
> > +        // Read the source routes file content
> > +        var routesFileContent = this.readFileAsString(routesFilePath);
> > +
> > +        // Append the new state
> > +        routesFileContent = routesFileContent.replace('$stateProvider.', engine(this.read('_.client.route.js'), this));
>
> Why do you need ejs?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meanjs/generator-meanjs/pull/165/files#r49158473>.
>

codydaig replied:
:+1:

ilanbiala commented on test/angular-view.test.js:
```diff
@@ -0,0 +1,197 @@
+          "        url: '/barroute',\n" +
+          "        templateUrl: 'modules/core/client/views/bar.client.view.html'\n" +
+          "      }).\n" +
+          "      state('core', {\n" +
```
If possible the dot should be on this line and not above

ilanbiala commented:
I think the tests definitely were a good idea, and thanks for adding them. @codydaig comments?

codydaig commented on test/angular-view.test.js:
```diff
@@ -0,0 +1,197 @@
+          "  function($stateProvider) {\n" +
+          "    // Core state routing\n" +
+          "    $stateProvider.\n" +
+          "      state('bar', {\n" +
```
If @ilanbiala's comment below on line 187 holds true, than it should be the same up here as well.

codydaig commented:
If possible, make the minor change that @ilanbiala pointed out with regards to the style guide formating of the `.`. Otherwise, LGTM. Thanks!

UndeadBaneGitHub commented:
Kk, will apply these changes today, and add likewise tests to the remaining
two branches too.

2016-01-08 17:55 GMT+03:00 Cody B. Daig <notifications@github.com>:

> If possible, make the minor change that @ilanbiala
> <https://github.com/ilanbiala> pointed out with regards to the style
> guide formating of the .. Otherwise, LGTM. Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meanjs/generator-meanjs/pull/165#issuecomment-170023440>
> .
>

UndeadBaneGitHub commented:
@codydaig @ilanbiala  Ok, here I was able to squash the commits into one, but then I had to add an additional merge afterwards, because this branch affects the gulpfile.js. I hope, that's ok.

codydaig commented:
@ilanbiala Thoughts on the second commit? I'm okay with letting it go this time sine squashing them down isn't an option.

ilanbiala commented:
@UndeadBaneGitHub did you make any changes to these files in this PR or is that just a result of them being added to master after this PR was made? What doesn't work with squashing the merged commit as a fixup?

UndeadBaneGitHub commented:
gulpfile.js and package.json were changed, an if I don't merge, they are
marked as conflicts.

On Wednesday, 13 January 2016, Ilan Biala <notifications@github.com> wrote:

> @UndeadBaneGitHub <https://github.com/UndeadBaneGitHub> did you make any
> changes to these files in this PR or is that just a result of them being
> added to master after this PR was made? What doesn't work with squashing
> the merged commit as a fixup?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meanjs/generator-meanjs/pull/165#issuecomment-171115859>
> .
>

ilanbiala commented:
@UndeadBaneGitHub what about squashing the 2 commits down to one?

UndeadBaneGitHub commented:
Tried that - gets conflicted once again :)

On Wednesday, 13 January 2016, Ilan Biala <notifications@github.com> wrote:

> @UndeadBaneGitHub <https://github.com/UndeadBaneGitHub> what about
> squashing the 2 commits down to one?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meanjs/generator-meanjs/pull/165#issuecomment-171125775>
> .
>

UndeadBaneGitHub commented:
Maybe, I just create a new PR instead of this one?

On Wednesday, 13 January 2016, Vasily Ovchinnikov <u9262450174@gmail.com>
wrote:

> Tried that - gets conflicted once again :)
>
> On Wednesday, 13 January 2016, Ilan Biala <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> @UndeadBaneGitHub <https://github.com/UndeadBaneGitHub> what about
>> squashing the 2 commits down to one?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/meanjs/generator-meanjs/pull/165#issuecomment-171125775>
>> .
>>
>

ilanbiala commented:
Yeah sure, feel free if that is easier.

UndeadBaneGitHub commented:
Created a new PR https://github.com/meanjs/generator-meanjs/pull/182, closing this one.
